src/store/app.ts

```diff
@@ -206,6 +231,53 @@ export const useAppStore = defineStore('app', {
                 return false
             }
         },
+        /**
+         * Asynchronously retrieves the shopping cart of the current user.
+         *
+         * @returns A promise that resolves to true if the shopping cart was successfully restored, false otherwise.
+         */
+        async restoreShoppingCartOfCurrentUser(): Promise<boolean> {
+            if (this.currentUserId != undefined) {
+                const getShoppingCartOfUserQuery =
+                    await awaitActionAndPushErrorIfNecessary(() => {
+                        return useClient().getShoppingCartOfUser({
+                            id: this.currentUserId,
+                        })
+                    }, errorMessages.restoreShoppingCartOfUser)
+
+                const shoppingCartOfCurrentUser =
+                    getShoppingCartOfUserQuery.user.shoppingcart
+                this.shoppingCart.lastUpdatedAt =
+                    shoppingCartOfCurrentUser.lastUpdatedAt
+                this.shoppingCart.items = []
+
+                shoppingCartOfCurrentUser.shoppingcartItems.nodes.forEach(
```

Comment:
soo, Becker has this 20 LOC rule for function bodies
while I guess this rule is not that strict for frontend dev, this is still a bit long
maybe this foreach block in the middle can be refactored into another function?

Reply:
5d5a562aea9175c9934346122284a7d9950a6048